torchtitan/experiments/kernels/moe/dispatch.py

```diff
@@ -0,0 +1,318 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import torch
+import torch.distributed as dist
+import torch.distributed._symmetric_memory as symm_mem
+
+
+# Adding out-of-tree ops to the `symm_mem` library
+lib = torch.library.Library("symm_mem", "FRAGMENT")  # noqa: TOR901
+
+"""
+all_to_all_vdev_2d_offset_copy:
+Copy data from `input` to `symm_in_buf` and call `all_to_all_vdev_2d_offset` to shuffle data
+"""
+lib.define(
+    "all_to_all_vdev_2d_offset_copy("
+    "Tensor input, Tensor symm_in_buf, Tensor(a!) out, "
+    "Tensor in_splits_offsets, Tensor(a!) out_splits_offsets, str group_name) -> ()",
+    tags=[torch._C.Tag.needs_fixed_stride_order],
```

Comment:
Should use `needs_exact_strides`, you will run into surprises with `needs_fixed_stride_order`

Reply:
all of these should be `needs_exact_strides`. `needs_fixed_stride_order` is broken lol. See https://github.com/pytorch/pytorch/blob/3a65ff84b69f2b255316732cab3e3acac4188f39/aten/src/ATen/native/tags.yaml#L45-L60 for descriptions